src/Logger_int.java

```diff
@@ -0,0 +1,340 @@
+
+        public void complete();
+
+        public void close();
+    }
+
+    public static class DummyProgressBar implements ProgressBar {
+
+        @Override
+        public void done(final int count) {
+        }
+
+        @Override
+        public void update() {
+        }
+
+        @Override
+        public void complete() {
+        }
+
+        @Override
+        public void close() {
+        }
+    }
+
+    public class PBar extends Thread implements ProgressBar {
+        private final Object _message;
+        private final int _total;
+        private final AtomicInteger _counter;
+        private volatile boolean _closeOnComplete;
+        private volatile boolean _update;
+        private int _lastMessageLen;
+
+        public PBar(final Object message, final int total,
+                final boolean closeOnComplete) {
+            _message = message;
+            _total = total;
+            _closeOnComplete = closeOnComplete;
+            _counter = new AtomicInteger(total);
+        }
+
+        @Override
+        public void run() {
+            for (int prev = 0; !isInterrupted();) {
+                final int current = _counter.get();
+
+                if (current <= 0) {
+                    break;
+                } else if (!_update && (current == prev)) {
+                    LockSupport.park();
+                } else {
+                    prev = current;
+                    _update = false;
+
+                    synchronized (Logger.this) {
+                        clearPogress();
+                        printProgress();
+                    }
+                }
+            }
+
+            synchronized (Logger.this) {
+                clearPogress();
+                printProgress();
+                clearPogress();
+                _pbar.remove(this);
+
+                if (!_closeOnComplete) {
+                    printMessage();
+                    _logWriter.println();
+                }
+
+                printProgress();
+            }
+        }
+
+        @Override
+        public void done(final int count) {
+            for (;;) {
+                final int current = _counter.get();
+
+                if (current <= 0) {
+                    return;
+                }
+
+                final int diff = current - count;
+
+                if (diff < 0) {
+                    _counter.set(0);
+                    LockSupport.unpark(this);
+                } else if (_counter.compareAndSet(current, diff)) {
+                    LockSupport.unpark(this);
+                    return;
+                }
+            }
+        }
+
+        @Override
+        public void update() {
+            _update = true;
+            LockSupport.unpark(this);
+        }
+
+        @Override
+        public void complete() {
+            _counter.set(0);
+            LockSupport.unpark(this);
+            try {
+                join();
+            } catch (final InterruptedException ex) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS2142)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS00108)